jscomp/others/js_re.mli

```diff
@@ -40,9 +40,29 @@
  * but WITHOUT ANY WARRANTY; without even the implied warranty of
  * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
  * GNU Lesser General Public License for more details.
- * 
+ *
  * You should have received a copy of the GNU Lesser General Public License
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA. *)
 
 type t
+type result
+
+external matches : result -> string array = "%identity"
+external index : result -> int = "" [@@bs.get]
+external input : result -> string = "" [@@bs.get]
+
+external fromString : string -> t = "RegExp" [@@bs.new]
+external fromStringWithFlags : string -> string -> t = "RegExp" [@@bs.new]
+
+external flags : t -> string = "" [@@bs.get]
+external global : t -> Js.boolean = "" [@@bs.get]
+external ignoreCase : t -> Js.boolean = "" [@@bs.get]
+external lastIndex : t -> int = "" [@@bs.get]
+external multiline : t -> Js.boolean = "" [@@bs.get]
+external source : t -> string = "" [@@bs.get]
+external sticky : t -> Js.boolean = "" [@@bs.get]
+external unicode : t -> Js.boolean = "" [@@bs.get]
+
```

Comment:
Oh, and it's not been added yet, you're just asking my opinion? It would certianly be nice for writing lots of bindings, but it's also more syntax to learn and will make it harder for beignners to read. So I think it should be avoided unless there's good documentation.